### ISSUE TITLE
Fix default config file paths on Windows (Trac #12)

### ISSUE THREAD

**Issue by jamesog** [jamesog/boxbackup#12]
The [wiki:KnownBugsWin32#DefaultConfiguration Known Bugs] page says:

"We should either always use the default hard-coded path (C:\Program Files\Box Backup\bbackupd.conf), or always assume that the configuration file is in the same directory as the executable, or always try both in the same order."

Migrated from https://www.boxbackup.org/ticket/12
```json
{
    "status": "closed", 
    "changetime": "2007-07-26T22:11:03", 
    "description": "The [wiki:KnownBugsWin32#DefaultConfiguration Known Bugs] page says:\n\n\"We should either always use the default hard-coded path (C:\\Program Files\\Box Backup\\bbackupd.conf), or always assume that the configuration file is in the same directory as the executable, or always try both in the same order.\"", 
    "reporter": "chris", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1185487863000000", 
    "component": "bbackupd", 
    "summary": "Fix default config file paths on Windows", 
    "priority": "normal", 
    "keywords": "configuration file location directory windows", 
    "version": "0.10", 
    "time": "2006-11-26T16:36:18", 
    "milestone": "0.11", 
    "owner": "chris", 
    "type": "defect"
}
```

**Comment by jamesog** [comment]
Trac update at 20070406T13:40:57: chris commented:

> Dave Bamford notes that bbackupd and bbackupquery check different paths, or in a different order, for the configuration file.

**Comment by jamesog** [comment]
Trac update at 20070526T15:33:27: 
* chris changed resolution from "" to "fixed"
* chris commented:

> (In [1690]) Determine config file location at runtime instead of hard-coding on
> Windows. Thanks to Paul MacKenzie, Per Thomsen, Pete Jalajas, Stuart
> Sanders, Dave Bamford and Gary for pushing me to do this. (fixes #12)


* chris changed status from "new" to "closed"

**Comment by jamesog** [comment]
Trac update at 20070526T17:35:54: chris commented:

> (In [1693]) New win32 native build (1692) with the following fixes and new features:
> 
>  * Fix saving and restoring archived state, broken in 1662, thanks Gary!
>    [1672]
> 
>  * Fix object use after delete, could crash client, thanks Gary! [1673]
> 
>  * Fix regular expression support to work on all platforms, and disabuse
>    HAVE_REGEX_H [1677] [1678] [1682]
> 
>  * Internal cleanups, pass std::strings to various classes instead of
>    char arrays [1683] [1684] [1689]
> 
>  * All programs use the directory of the executable as the default config
>    file location, instead of the hardcoded "C:\Program Files\Box Backup".
>    Thanks to Paul MacKenzie, Per Thomsen, Pete Jalajas, Stuart Sanders,
>    Dave Bamford and Gary for pushing me to do this. (fixes #12)
>    [1685] [1686] [1687] [1688] [1690] [1691] [1692]
> 
>  * Update copyrights to 2007, thanks Per Thomsen! [1670]

**Comment by jamesog** [comment]
Trac update at 20070726T22:11:03: chris commented:

> (In [1777]) Make Configuration take a std::string filename instead of a char array,
> in C++ style.
> 
> Add a function to get default config file paths at runtime, dependent on
> the location of the executable being run.
> 
> Pass the config file name directly to Daemon::Main, instead of faking argv.
> 
> No default raid file path at compile time on Windows, depends on
> executable location when run.
> 
> Determine RaidFile path at runtime if not supplied in config file
> on Windows.
> 
> Don't define default locations for config files at compile time on Windows,
> provide macros to determine them at runtime instead.
> 
> Make FileHandleGuard take a std::string instead of a char array, C++ style.
> 
> Determine config file location at runtime instead of hard-coding on
> Windows. Thanks to Paul MacKenzie, Per Thomsen, Pete Jalajas, Stuart
> Sanders, Dave Bamford and Gary for pushing me to do this. (fixes #12)
> 
> Determine config file path at runtime. Call Daemon::Main with config file
> name instead of building fake argv.
> 
> (#3, merges [1684] [1685] [1686] [1687] [1688] [1689] [1690]
> [1691] [1692])
